tests/templates/inject.html

```diff
@@ -1,5 +1,5 @@
 {% load component_tags %}
 <div>
-  {% component "injectee" %}
+  {% component "injectee17" %}
```

Comment:
As mentioned in https://github.com/django-components/django-components/issues/1413, there's now race conditions sometimes, as the finalizers for test A now run in parallel with the execution of test B, which sometimes causes problems.

As a simple workaround, I updated the tests where we rely on garbage collecion, so that each test defines a unique component name, so that the delayed finalizers aren't deleting incorrect entries.

I described the proper solution in https://github.com/django-components/django-components/issues/1413